### PR TITLE
fix: Show login modal when user session expires.

### LINKED ISSUE #2538: Expired session bug.
If the session expires while the user is editting their proposal, a "Forbidden" error message is displayed.

image

If I refresh the page, the error message goes away, but the submit button is still grayed out.

image

This UX does not give the user any information on what the problem is or how to correct it.  Shouldn't the login modal be displayed on these errors?

### PR DESCRIPTION
**Pull request by victorgcramos**
Close #2538

This diff fixes an issue that reported the absence of some login modal
when some user submits a proposal with an expired session.

### Dependencies
Depends on https://github.com/decred/pi-ui/pull/369

### UI Changes Screenshot

Screen Shot 2021-08-17 at 5 36 04 PM

### CODE REVIEW

**Comment by victorgcramos** [comment]
Improved this PR to comprise all possible expired sessions errors, by improving the `SessionChecker` HOC to display the login modal when some user session expires.

**Review by lukebp — CHANGES_REQUESTED** [review]
This currently displays the modal as soon as the session expires.  This is not quite what we want.  The modal should be displayed once the session expires AND the user attempts to access a route that requires the user to be logged in (ex. proposal submission, comment submission, etc).

**Comment by victorgcramos** [comment]
@lukebp handling all requests and displaying the login modal for them will require an extensive refactor because we need to change the way we handle auth errors on forms and other private actions. The Login modal display is an async procedure, so all 403 errors would be displayed in the form error message before the login modal shows up, just like other response errors.

I opted to go for this approach because this way we can avoid those syncing problems mentioned above. After talking to @tiagoalvesdulce, we both agreed that it should be included in the new architecture.

**Comment by lukebp** [comment]
Ok, that's fine.  We definitely don't want to display the login modal as soon as the session expires though.  We've done this before and it was very annoying for users because they would leave tabs open when reading proposals and would constantly have to log back in.

We have two options:

1. Hardcode the modal behavior to pop up when the session has expired for certain routes.  Probably just the proposal submission and comment submission route for now.

or 

2. Do nothing for now and wait for the new architecture.

Thoughts?
@tiagoalvesdulce @victorgcramos

**Comment by victorgcramos** [comment]
1 for me is the best.

**Comment by lukebp** [comment]
Ok, go with option 1 then.

**Comment by tiagoalvesdulce** [comment]
I'm OK with option 1.

**Review by lukebp — APPROVED** [review]
tACK

**Comment by lukebp** [comment]
@victorgcramos can you fix the merge conflicts?

This needs final approval from @tiagoalvesdulce.

**Comment by victorgcramos** [comment]
Let's wait for #2549 first.

**Review by amass01 — COMMENTED** [review]
Good job, tested and works fine.

I suggest we show the modal also on comment vote. 
Also, added a minor inline comment.

File: teste2e/cypress/e2e/comments/comments.js [hunk 1]
```diff
@@ -95,3 +95,29 @@ describe("User comments", () => {
...
+describe("Failed comments", () => {
+  let token = "";
+  beforeEach(() => {
+    const user = {
+      email: "adminuser@example.com",
+      username: "adminuser",
+      password: "password"
+    };
+    cy.login(user);
+    cy.intercept("/api/ticketvote/v1/inventory").as("inventory");
```
[the remark this answers is not in the document]
Comment (victorgcramos) [reply teste2e/cypress/e2e/comments/comments.js:108]: I'd wait until the mocked api structure is done to mock all requests. This is just an alias without any response changes.

[the remark this answers is not in the document]
Comment (amass01) [reply teste2e/cypress/e2e/comments/comments.js:108]: Sweet

**Review by amass01 — APPROVED** [review]
nice work, LGTM!

**Comment by lukebp** [comment]
Please update your commit message.  Same comment that I said here.

https://github.com/decred/politeiagui/pull/2568#issuecomment-920501890

**Pull request merged**